From: Brian C. Lane <bcl@redhat.com>
Subject: libparted: copy flags when duplicating GPT partitions
Origin: upstream, http://git.debian.org/?p=parted/parted.git;a=commitdiff;h=febeedd8d3d
Forwarded: not-needed
Last-Update: 2011-05-15

* libparted/labels/gpt.c (gpt_partition_duplicate): Copy flags to new
partition.

Reported by Chris Murphy in https://bugzilla.redhat.com/747497.

---
 libparted/labels/gpt.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/libparted/labels/gpt.c b/libparted/labels/gpt.c
index 8c895c4..dc3e8b4 100644
--- a/libparted/labels/gpt.c
+++ b/libparted/labels/gpt.c
@@ -1348,9 +1348,7 @@ gpt_partition_duplicate (const PedPartition *part)
   if (!result_data)
     goto error_free_part;
 
-  result_data->type = part_data->type;
-  result_data->uuid = part_data->uuid;
-  strcpy (result_data->name, part_data->name);
+  *result_data = *part_data;
   return result;
 
 error_free_part:
-- 
